services/github/github-commit-activity.service.js

```diff
@@ -1,29 +1,35 @@
+import gql from 'graphql-tag'
 import Joi from 'joi'
+import { InvalidResponse } from '../index.js'
 import { metric } from '../text-formatters.js'
 import { nonNegativeInteger } from '../validators.js'
-import { GithubAuthV3Service } from './github-auth-service.js'
-import { errorMessagesFor, documentation } from './github-helpers.js'
+import { GithubAuthV4Service } from './github-auth-service.js'
+import { transformErrors, documentation } from './github-helpers.js'
 
-const schema = Joi.array()
-  .items(
-    Joi.object({
-      total: nonNegativeInteger,
-    })
-  )
-  .required()
+const schema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      object: Joi.object({
+        history: Joi.object({
+          totalCount: nonNegativeInteger,
+        }).required(),
+      }),
+    }).required(),
+  }).required(),
+}).required()
 
-export default class GithubCommitActivity extends GithubAuthV3Service {
+export default class GitHubCommitActivity extends GithubAuthV4Service {
   static category = 'activity'
   static route = {
     base: 'github/commit-activity',
-    pattern: ':interval(y|m|4w|w)/:user/:repo',
+    pattern: ':interval(y|m|4w|w)/:user/:repo/:branch*',
   }
 
   static examples = [
     {
       title: 'GitHub commit activity',
       // Override the pattern to omit the deprecated interval "4w".
-      pattern: ':interval(y|m|w)/:user/:repo',
+      pattern: ':interval(y|m|w)/:user/:repo/',
```

Comment:
was debating whether to add another example that showed the ability to specify a branch/ref but that ties in to a broader question asked below